src/sqlfluff/rules/L064.py

```diff
@@ -0,0 +1,262 @@
+            and not self.force_enable
+        ):
+            return LintResult(memory=context.memory)
+
+        # Only care about quoted literal segments.
+        if not context.segment.name == "quoted_literal":
+            return None
+
+        # If quoting style is set to consistent we use the quoting style of the first
+        # quoted_literal that we encounter.
+        if self.preferred_string_quotes == "consistent":
+            memory = context.memory
+            preferred_string_quotes = memory.get("preferred_string_quotes")
+
+            if not preferred_string_quotes:
+                # Getting the quote from LAST character to be able to handle STRING
+                # prefixes
+                preferred_string_quotes = (
+                    "double_quotes"
+                    if context.segment.raw[-1] == '"'
+                    else "single_quotes"
+                )
+                memory["preferred_string_quotes"] = preferred_string_quotes
+                self.logger.debug(
+                    "Preferred string quotes is set to `consistent`. Derived quoting "
+                    "style %s from first quoted literal.",
+                    preferred_string_quotes,
+                )
+        else:
+            preferred_string_quotes = self.preferred_string_quotes
+
+        fixed_string = self._normalize_preferred_string_quotes(
+            context.segment.raw,
+            preferred_quote_char=QUOTES_MAPPING[preferred_string_quotes][
+                "preferred_quote_char"
+            ],
+            alternate_quote_char=QUOTES_MAPPING[preferred_string_quotes][
+                "alternate_quote_char"
+            ],
+        )
+
+        if fixed_string != context.segment.raw:
+            return LintResult(
+                anchor=context.segment,
+                memory=context.memory,
+                fixes=[
+                    LintFix.replace(
+                        context.segment,
+                        [
+                            CodeSegment(
+                                raw=fixed_string,
+                                name="quoted_literal",
+                                type="literal",
+                            )
+                        ],
+                    )
+                ],
+                description=(
+                    "Inconsistent use of preferred quote style '"
+                    f"{QUOTES_MAPPING[preferred_string_quotes]['common_name']}'. "
+                    f"Use {fixed_string} instead of {context.segment.raw}."
+                ),
+            )
+
+        return None
+
+    def _sub_twice(self, regex: regex.Pattern, replacement: str, original: str) -> str:
+        """Replace `regex` with `replacement` twice on `original`.
+
+        This is used by string normalization to perform replaces on overlapping matches.
+        """
+        return regex.sub(replacement, regex.sub(replacement, original))
```

Comment:
I'm not getting what this is for? And why you need to do this twice (and only twice)?

My Python is poor btw, so maybe this is obvious to a real Python developer.

Reply:
I have to admit I only understand 80% of the regex magic that is done in the implementation. I just copied this. It's talking about overlapping matches, so I'm prepared to just believe that this is actually needed 😂

Reply:
In that case, can we include a comment/link to the code it was copied from?